src/Raven.Server/Documents/Handlers/RevisionsHandler.cs

```diff
@@ -81,6 +82,31 @@ public Task GetRevisionsFor()
             }
         }
 
+        [RavenAction("/databases/*/revisions/revert", "GET", AuthorizationStatus.ValidUser)]
+        public async Task Revert()
+        {
+            var time = GetDateTimeQueryString("point-in-time");
+            var window = GetTimeSpanQueryString("window", required: false) ?? TimeSpan.FromHours(48);
+
+            var operationId = GetLongQueryString("operationId", required: false) ?? Database.Operations.GetNextOperationId();
+            var token = CreateOperationToken();
+
+            try
+            {
+                await Database.Operations.AddOperation(
+                    Database,
+                    $"Revert database '{Database.Name}' to {time}.",
+                    Operations.Operations.OperationType.DatabaseRevert,
+                    onProgress => Task.Run(() => Database.DocumentsStorage.RevisionsStorage.RevertRevisions(time.Value, window, onProgress), token.Token), 
+                    operationId, 
+                    token: token);
+            }
+            catch (Exception)
+            {
+                HttpContext.Abort();
```

Comment:
What about sending the error to the user? Here you just disconnect the connection, which is probably not a good idea.